api/src/com/cloud/network/Networks.java

```diff
@@ -251,6 +252,10 @@ public static URI fromString(String candidate) {
                 if (com.cloud.dc.Vlan.UNTAGGED.equalsIgnoreCase(candidate)) {
                     return Native.toUri(candidate);
                 }
+                if (UuidUtils.validateUUID(candidate)){
+                    //Supporting lswitch uuid as vlan id: set broadcast_uri = null and add row on nicira_nvp_router_map with logicalrouter_uuid = candidate
+                    return null;
```

Comment:
The idea is to support creating Shared Networks over NSX Managed Network Offering, so there will be this two cases (for creation):
* Providing numerical VLAN_ID:
 * broadcastdomain_type = Lswitch
 * broadcast_uri = "vlan://VLAN_ID"
* Providing logical router UUID as VLAN_ID:
 * broadcastdomain_type = Lswitch
 * broadcast_uri = NULL
 * add entry on nicira_nvp_router_map with network id and logical router's UUID

For second case, I included the code in the catch block as I asumed that VLAN_ID provided is not numerical (as long parsing fail).